File: back/backend/src/main/java/com/foro_cine/backend/post/Post.java
package com.foro_cine.backend.post;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "posts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titulo;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String contenido;

    @Column(nullable = false)
    private String autor;

    @Column(nullable = false)
    private String fecha;

    private int likes;
    private int dislikes;
}

File: frontend/app/src/main/java/com/example/foro_cinev1/domain/models/Post.kt
package com.example.foro_cinev1.domain.models

data class Post(
    val id: Int = 0,
    val titulo: String,
    val contenido: String,
    val autor: String,
    val fecha: String,
    val likes: Int = 0,
    val dislikes: Int = 0,
    var userVote: Int = 0
)
